### ISSUE TITLE
Test CI with spack test run

### ISSUE THREAD

**Issue by janciesko** [kokkos/kokkos#4696]
Add `spack test run kokkos` to our CI.

**Comment by Rombur** [comment]
What should we add to the CI exactly? Do we pull the development version of spack and install the development version of Kokkos with it? Which configuration should we test?

**Comment by ajpowelsnl** [comment]
Morning @janciesko and @Rombur -- Do you expect that the new Spack capability will be in Kokkos 3.6?

**Comment by janciesko** [comment]
Yes, @Rombur, development version of both. If we see that either is failing, we know early and can make fixes for releases. I think one combo of device and host back-end should be good enough to test spack. All other combos are covered elsewhere.
